### PR TITLE
add synthesis60

### PR DESCRIPTION
**Pull request by dayatz**
# synthesis60

ergo layout pcb that fits to most alice-like boards
pcb

* Keyboard Maintainer: [dayatz](https://github.com/dayatz)
* Hardware Supported: synthesis60, ATMega32u4
* Hardware Availability: Local Indonesia GB.

Make example for this keyboard (after setting up your build environment):

    make dyz/synthesis60:default

Flashing example for this keyboard:

    make dyz/synthesis60:default:flash

See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).

### CODE REVIEW

File: keyboards/dyz/synthesis60/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+void oled_task_user(void) {
+    // Host Keyboard Layer Status
+    oled_write_P(PSTR("Synthesis60\nLayer: "), false);
+    // Host Keyboard LED Status
+    led_t led_state = host_keyboard_led_state();
+    oled_write_P(led_state.num_lock ? PSTR("NUM ") : PSTR("    "), false);
+    oled_write_P(led_state.caps_lock ? PSTR("CAP ") : PSTR("    "), false);
+    oled_write_P(led_state.scroll_lock ? PSTR("SCR ") : PSTR("    "), false);
+}
+#endif
```
Comment (drashna) [inline keyboards/dyz/synthesis60/keymaps/default/keymap.c:58]: This should be moved to the keyboard's c file. 
```suggestion
```

File: keyboards/dyz/synthesis60/keymaps/via/keymap.c [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+void oled_task_user(void) {
+    // Host Keyboard Layer Status
+    oled_write_P(PSTR("Synthesis60\nLayer: "), false);
+    // Host Keyboard LED Status
+    led_t led_state = host_keyboard_led_state();
+    oled_write_P(led_state.num_lock ? PSTR("NUM ") : PSTR("    "), false);
+    oled_write_P(led_state.caps_lock ? PSTR("CAP ") : PSTR("    "), false);
+    oled_write_P(led_state.scroll_lock ? PSTR("SCR ") : PSTR("    "), false);
+}
+#endif
```
Comment (drashna) [inline keyboards/dyz/synthesis60/keymaps/via/keymap.c:58]:
```suggestion
```

File: keyboards/dyz/synthesis60/rules.mk [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (drashna) [inline keyboards/dyz/synthesis60/rules.mk:11]:
```suggestion
```

File: keyboards/dyz/synthesis60/rules.mk [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+ENCODER_ENABLE = no
+OLED_DRIVER_ENABLE = yes
```
Comment (drashna) [inline keyboards/dyz/synthesis60/rules.mk:28]:
```suggestion
NKRO_ENABLE = yes           # USB Nkey Rollover
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
ENCODER_ENABLE = no
OLED_ENABLE = yes
```

File: keyboards/dyz/synthesis60/synthesis60.c [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "synthesis60.h"
```
Comment (drashna) [inline keyboards/dyz/synthesis60/synthesis60.c:16]:
```suggestion
#include "synthesis60.h"

#ifdef OLED_ENABLE
bool oled_task_kb(void) {
    if (!oled_task_kb()) { return false; }
    // Host Keyboard Layer Status
    oled_write_P(PSTR("Synthesis60\nLayer: "), false);
    // Host Keyboard LED Status
    led_t led_state = host_keyboard_led_state();
    oled_write_P(led_state.num_lock ? PSTR("NUM ") : PSTR("    "), false);
    oled_write_P(led_state.caps_lock ? PSTR("CAP ") : PSTR("    "), false);
    oled_write_P(led_state.scroll_lock ? PSTR("SCR ") : PSTR("    "), false);

    return true;
}
#endif
```

**Review by drashna — CHANGES_REQUESTED** [review]
> ☒ dyz/synthesis60: Number of keys for LAYOUT_all does not match! info.json specifies 66 keys, C macro specifies 69

**Review by drashna — APPROVED** [review]

File: keyboards/dyz/synthesis60/rules.mk [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+#
+BOOTMAGIC_ENABLE = yes     # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = no       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no        # Commands for debug and configuration
+NKRO_ENABLE = yes           # USB Nkey Rollover
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+ENCODER_ENABLE = no
+OLED_ENABLE = yes
```
Comment (fauxpark) [inline keyboards/dyz/synthesis60/rules.mk:18]:
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
NKRO_ENABLE = yes           # Enable N-Key Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
ENCODER_ENABLE = no
OLED_ENABLE = yes
```

File: keyboards/dyz/synthesis60/info.json [hunk 7]
```diff
@@ -0,0 +1,123 @@
+{
```
Comment (fauxpark) [inline keyboards/dyz/synthesis60/info.json:1]: This file needs to be redone without the rotation:
image

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**